MC/Geant4Config.C

```diff
@@ -58,7 +58,7 @@ void Geant4Config()
       runConfiguration->SetParameter("monopoleMagCharge", mCh);
       
       geant4 = new TGeant4("TGeant4", 
-			   Form("The Geant4 Monte Carlo : %s+monopole", configPhysList.Data()), 
+			   Form("The Geant4 Monte Carlo : %s+monopole, EMV-EMCAL", configPhysList.Data()), 
 			   runConfiguration);
```

Comment:
The v3-6-p6-inclxx-biasing-p2 does not include "monopole" physics. This development is available since v4-0 and it was back-ported to v3-2-patches (which is older than v3-6-...), as this version was deployed by ALICE at that time. Do you need  "monopole" physics for the https://alice.its.cern.ch/jira/browse/ALIROOT-8401 studies ?

Reply:
https://alice.its.cern.ch/jira/browse/ALIROOT-8401 does not need monopoles. Which is why I commented in the ticket that we should move to a new build, with v3-6-p6-inclxx-biasing-p2 if @fprino , @gconesab , @catalinristea agree too.

Reply:
But now I wonder: don't we need a protection for the Physics List? If the default is BERT+biasing, and not all versions have it, especially old builds, they will fail. Should we check the G4 version, then, in the Geant4Config.C macro?
Same for the monopoles...

Reply:
The Geant4 VMC version can be accessed in g4Config.C macro in this way:
```
#include "TG4Version.h"
 cout << "Geant4 VMC version: " <<  GEANT4_VMC_RELEASE << endl;
```
what gives me:
`Geant4 VMC version: 5.1.p1
`
The monopole physics is available since 4.0, the biasing since 5.0 and the old versions, where these development were back-ported contain the string "monopole" or "biasing"  (the complete strings are "3.2.3.monopole" and "3.6.p6.inclxx-biasing-p2").

Reply:
Hello Ivana, 

The config macro is not compiled, the include would have no effect. Or you tried in within alidpg and it worked?
An alternative is to use the packages from the JDL.

Chiara

Reply:
I tested with VMC example and it was needed here. I did not try with AliDPG.

Reply:
Ciao Ivana,

Trying with locally installed aliroot a simple test macro, I get:
```
root [0] .x testG4.C 
Error: cannot open file "TG4Version.h"  testG4.C:2:
*** Interpreter error recovered ***
root [1] 
```
Do you have special settings? Can you try on lxplus, with an analysis tag?

Reply:
For me it works on lxplus:
```
$ alienv enter VO_ALICE@AliPhysics::vAN-20200730_ROOT6-1
$ [AliPhysics/vAN-20200730_ROOT6-1] ~ > root.exe
   ------------------------------------------------------------------
  | Welcome to ROOT 6.20/02                        https://root.cern |
  | (c) 1995-2020, The ROOT Team; conception: R. Brun, F. Rademakers |
  | Built for linuxx8664gcc on Jun 09 2020, 17:02:00                 |
  | From tags/v6-20-02-alice6@v6-07-06-16910-g8289f4b                |
  | Try '.help', '.demo', '.license', '.credits', '.quit'/'.q'       |
   ------------------------------------------------------------------


WELCOME to ALICE

root [0] .x testG4.C 
Geant4 VMC version: 5.0.p3
root [1] 
```
Where testG4.C is:
```
#include "TG4Version.h"

void testG4()
{
  cout << "Geant4 VMC version: " <<  GEANT4_VMC_RELEASE << endl;
}

```

Reply:
Hello,

My macro is very similar (and called in the same way :-) I attach it below. 
I found the problem: the include is not recognized with ROOT5 builds, you can try on lxplus. 
So, I guess we should fall back to use the JDL packages. What do you (all) think?

Cheers,

Chiara 
----
my macro:
```
#include "TG4Version.h"
void testG4() {

TString g4libsMacro="$ALIDPG_ROOT/MC/g4libs.C";
  if (!gInterpreter->IsLoaded(g4libsMacro.Data())) {
      gROOT->LoadMacro(g4libsMacro.Data());
      gInterpreter->ProcessLine("g4libs()");
  }
  TG4RunConfiguration* runConfiguration = NULL;
  TString configPhysList = "FTFP_INCLXX_EMV+optical";
  runConfiguration = new TG4RunConfiguration("geomRoot", Form("%s", configPhysList.Data()), "specialCuts+stackPopper+stepLimiter", true);
  
  //TGeant4 *geant4 = new TGeant4("TGeant4", Form("The Geant4 Monte Carlo : %s, EMV-EMCAL", configPhysList.Data()), runConfiguration);

  cout << "Geant4 VMC version: " <<  GEANT4_VMC_RELEASE << endl;

  return;

}
```

Reply:
With Root 5, you need to add the path to the interpreter; to make the include work in the testG4.C macro, it has to be done before this macro call:
```
root [0] .x addG4Path.C
root [1] .x testG4.C 
Geant4 VMC version: 3.6.p3

```
Where addG4Path.C:
```
void addG4Path()
{
  gInterpreter->AddIncludePath("$G4VMCINSTALL/include/geant4vmc");  
}
```
The env variable G4VMCINSTALL is defined via alienv.

But you can of course move the check higher and implement it at the JDL level.

Reply:
Ciao @ihrivnac , 

I confirm that like that it works. 
I will then follow your suggestion. 

Cheers,

Chiara

Reply:
Hello @ihrivnac , @gconesab ,

I implemented what Ivana suggested. Can you take a look?
Note that we cannot use the JDL, since G4 is not explicitly mentioned in the packages, so this is the only way.
I have a question: doesn't G4 abort if you call it with a Physics List that it does not recognize? This would make the check on the version a bit redundant, but we could have clearer logs in some cases.
I also wonder about the deletion of geant4 here https://github.com/alisw/AliDPG/blob/9a4fc651c1d8d671e3f364bfcf12d7a3c27fb743/MC/Geant4Config.C#L39 (line 39 of master):

```
    else{
      Printf("\n\nSET UP GEANT4 for magnetic monopoles\n\n");
      if (g4PhysList == "BERT+biasing") {
	Printf("We cannot use the BERT+biasing with the monopoles, we will unset the pointer to geant4");
	delete geant4;
	geant4 = 0;
```
I don't think it is actually needed, "geant4" is not created yet at this line, no?

@gconesab , could you test this with both ROOT5 and ROOT6, if this approach is considered satisfactory?

Cheers,

Chiara

Reply:
Hi @chiarazampolli ,
Geant4 VMC aborts when the physics list is not recognized; with the message below:

```
    TG4RunConfiguration::TG4RunConfiguration:
    Physics list selection FTFP_BERT+abc not recognized.
    Available options:
    EMonly, EMonly+Extra, Hadron_EM, Hadron_EM+Extra
      where EMonly = emStandard 
            Hadron = FTFP_BERT FTFP_BERT_TRV FTFP_BERT_ATL FTFP_BERT_HP FTFQGSP_BERT FTFP_INCLXX FTFP_INCLXX_HP FTF_BIC LBE QBBC QGSP_BERT QGSP_BERT_HP QGSP_BIC QGSP_BIC_HP QGSP_BIC_AllHP QGSP_FTFP_BERT QGSP_INCLXX QGSP_INCLXX_HP QGS_BIC Shielding ShieldingLEND ShieldingM NuBeam 
            EM =  _EMV _EMX _EMY _EMZ _LIV _PEN __GS __SS _EM0 _WVI __LE 
            Extra = biasing extra monopole optical radDecay 
      The Extra selections are cumulative, while Hadron selections are exlusive.
    
*** TG4Exception: Aborting execution ***

```

Reply:
Exactly, so the checks are just to avoid that it aborts, but in principle, a mismatch between version and physics list would not work by default.